src/helpers/helpers.collection.ts

```diff
@@ -2,16 +2,20 @@ import {_capitalize} from './helpers.core';
 
 /**
  * Binary search
- * @param {array} table - the table search. must be sorted!
- * @param {number} value - value to find
- * @param {function} [cmp]
+ * @param table - the table search. must be sorted!
+ * @param value - value to find
+ * @param cmp
  * @private
  */
-export function _lookup(table, value, cmp) {
+export function _lookup(
+  table: unknown[],
```

Comment:
I think it might be good to change this definition slightly to

```ts
export function _lookup<T>(
  table: T[],
  value: number,
  cmp?: (value: T) => boolean
)
```

My thinking is that we default to assuming `T` is a number on line 15 so this will guard against someone passing a complex object but forgetting to pass the `cmp` function.

We could also do this with an override so there is one version where you pass `number[]` and no `cmp` and then if you pass an object, `cmp` becomes required

Reply:
done